app/tray/Account/Requests/SignPermitRequest/index.js

```diff
@@ -0,0 +1,261 @@
+import React, { useState, useEffect } from 'react'
+import BigNumber from 'bignumber.js'
+
+import { formatDisplayInteger, isUnlimited } from '../../../../../resources/utils/numbers'
+import svg from '../../../../../resources/svg'
+import link from '../../../../../resources/link'
+import { ClusterBox, Cluster, ClusterRow, ClusterValue } from '../../../../../resources/Components/Cluster'
+import Countdown from '../../../../../resources/Components/Countdown'
+import RequestHeader from '../../../../../resources/Components/RequestHeader'
+import RequestItem from '../../../../../resources/Components/RequestItem'
+import CustomAmountInput from '../../../../../resources/Components/CustomAmountInput'
+import TypedSignatureOverview from '../../../../../resources/Components/SimpleTypedData'
+import useCopiedMessage from '../../../../../resources/Hooks/useCopiedMessage'
+
+import { getSignatureRequestClass } from '../../../../../resources/domain/request'
+const getPermit = (req) => {
+  const {
+    typedMessage: {
+      data: {
+        message: { deadline, spender, value, owner, nonce },
+        domain: { verifyingContract, chainId, name, version },
+        types
+      }
+    }
+  } = req
+
+  return {
+    deadline,
+    spender,
+    value,
+    owner,
+    verifyingContract,
+    chainId,
+    name,
+    nonce,
+    version,
+    types
+  }
+}
+
+const PermitOverview = ({ permit, req, chainData, originName, tokenData }) => {
+  const { owner, deadline, spender, value } = permit
+  const { chainColor, chainName, icon } = chainData
+  const [showCopiedMessage, copySpender] = useCopiedMessage(spender)
+
+  //TODO: allow time limit & value to be edited...
+  return (
+    <div className='approveRequest'>
+      <div className='approveTransactionPayload'>
+        {/* //TODO: loading state when getting token data */}
```

Comment:
are these TODOs still outstanding? if so maybe turn them into Notion tickets so we can get rid of the comments before committing